src/tiled/automapper.cpp

```diff
@@ -749,7 +797,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 if (listNo.isEmpty()) {
                     if (matchListYes)
                         continue;
-                    if (!ruleDefinedListYes && !cells.contains(c1))
```

Comment:
Indeed, maybe the best would be to have a property to handle the way it works when there is no inputnot_layer.
Otherwise, we'd have to add an empty inputnot_layer for each input_layer.